tests/maddy/test_maddy_strongparser.cpp

```diff
@@ -83,3 +83,118 @@ TEST(MADDY_STRONGPARSER, ItDoesNotParseInsideInlineCode)
     ASSERT_EQ(test.expected, test.text);
   }
 }
+
+TEST(MADDY_STRONGPARSER, ItReplacesUnderscoresAtStringEdges)
+{
+  std::string text = "__some text__";
+  std::string expected = "<strong>some text</strong>";
+  auto strongParser = std::make_shared<maddy::StrongParser>();
+
+  strongParser->Parse(text);
+
+  ASSERT_EQ(expected, text);
+}
+
+TEST(DISABLED_MADDY_STRONGPARSER, ItDoesNotReplaceMarkdownWithInlineUnderscores)
+{
+  std::string text = "some text__bla__text testing __it__ out";
+  std::string expected = "some text__bla__text testing <strong>it</strong> out";
+  auto strongParser = std::make_shared<maddy::StrongParser>();
+
+  strongParser->Parse(text);
+
+  ASSERT_EQ(expected, text);
+}
+
+TEST(DISABLED_MADDY_STRONGPARSER, ItOnlyReplacesUnderscoresAtWordBreaks)
+{
+  std::string text = "some __text__bla__ testing __it__ out";
+  std::string expected =
+    "some <strong>text__bla</strong> testing <strong>it</strong> out";
+  auto strongParser = std::make_shared<maddy::StrongParser>();
+
+  strongParser->Parse(text);
+
+  ASSERT_EQ(expected, text);
+}
+
+TEST(MADDY_STRONGPARSER, ItReplacesUnderscoresWithMultipleWords)
+{
+  std::string text = "some __text testing it__ out";
+  std::string expected = "some <strong>text testing it</strong> out";
+  auto strongParser = std::make_shared<maddy::StrongParser>();
+
+  strongParser->Parse(text);
+
+  ASSERT_EQ(expected, text);
+}
+
+TEST(DISABLED_MADDY_STRONGPARSER, ItAllowsTripleUnderscores)
+{
+  // I'm not sure if this is standard or not, but this is how the github
+  // markdown parser behaves.  Other things I've seen want it to *not*
+  // match.
+
+  std::string text = "some ___text testing it__ out";
+  std::string expected = "some <strong>_text testing it</strong> out";
```

Comment:
In the CommonMark Spec, there is this example:

```
markdown	"___foo__\n"
html	"<p>_<strong>foo</strong></p>\n"
example	456
```
Also GitHub spec: https://github.github.com/gfm/#example-465

So I guess, the `_` in the expected string has to come before the strong tag.